### PR TITLE
feat: make changes to session manager to work with RN

### PR DESCRIPTION
**Pull request by iykazrji**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

## Notes

---

## PR-Codex overview
This PR focuses on integrating `react-native-mmkv` for local storage management in a React Native application, enhancing the local storage capabilities and ensuring compatibility with existing session management.

### Detailed summary
- Added `react-native-mmkv` dependency for local storage.
- Implemented `MMKVLocalStorage` class in `mmkv-localstorage-polyfill.ts`.
- Set up global `localStorage` to use `MMKVLocalStorage` if not already defined.
- Updated `App.tsx` to test local storage functionality.
- Adjusted session management in `manager.ts` to check for `window` before adding focus event listener.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  clear() {
+    this.storage.clearAll();
+  }
+
+  get length() {
+    return this.storage.getAllKeys().length;
+  }
+
+  key(index: number) {
+    const keys = this.storage.getAllKeys();
```
Comment (moldy530) [inline account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts:33]: I'm guessing `getAllKeys` returns a key value pair array?

File: account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    return keys[index] || null;
+  }
+}
+
+if (typeof window !== "undefined" && !global.localStorage) {
```
Comment (moldy530) [inline account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts:38]: isn't window always undefined on mobile?

should this just check for `global.localStorage`?

File: account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts (hunk 1 above)
Comment (iykazrji) [reply account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts:33]: It returns an array of just the available keys, without their values. Here's from their docs: https://github.com/mrousavy/react-native-mmkv?tab=readme-ov-file#keys

File: account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts (hunk 2 above)
Comment (iykazrji) [reply account-kit/rn-signer/src/utils/mmkv-localstorage-polyfill.ts:38]: Fair point! Would be updating this

**Pull request closed without merging**